File: web/src/app/core/models/transaction.ts
import { Category } from './category';

export class Transaction {
    id: number;
    amount: number;
    category: Category;
    date: Date;
    description: string;
    isIncome: boolean;
    userId: string;

    constructor(data: any) {
        Object.assign(this, data);
        this.date = new Date(data.date);
    }
}

File: src/Domain/Entities/Transaction.cs
﻿using System;
using Domain.Common;
using Domain.Enums;

namespace Domain.Entities
{
    public class Transaction : Entity<long>
    {
        public Guid CategoryId { get; set; }
        public Guid UserId { get; set; }
        public decimal Amount { get; set; }
        public TransactionType Type { get; set; }
        public DateTime Date { get; set; }
        public string Description { get; set; }

        public Category Category { get; set; }
        public User User { get; set; }
    }
}
